pkg/virt-handler/non-root.go

```diff
@@ -106,18 +107,19 @@ func (d *VirtualMachineController) prepareStorage(vmi *v1.VirtualMachineInstance
 }
 
 func getTapDevices(vmi *v1.VirtualMachineInstance) []string {
-	macvtap := map[string]bool{}
+	macvtap := map[string]struct{}{}
 	for _, inf := range vmi.Spec.Domain.Devices.Interfaces {
 		if inf.Macvtap != nil {
-			macvtap[inf.Name] = true
+			macvtap[inf.Name] = struct{}{}
 		}
 	}
 
+	networkNameScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
 	tapDevices := []string{}
 	for _, net := range vmi.Spec.Networks {
-		_, ok := macvtap[net.Name]
-		if ok {
-			tapDevices = append(tapDevices, net.Multus.NetworkName)
+		_, isMacvtapNetwork := macvtap[net.Name]
+		if podInterfaceName, exists := networkNameScheme[net.Name]; isMacvtapNetwork && exists {
```

Comment:
What is the case for not `exist`?

Reply:
It shouldn't happen at all since we're iterating the networks, and this `networkNameScheme` is supposed to map the networks to pod interfaces.

But, I've seen weirder things. Maybe adding a log when it does **not** exist ? ... 

But still: it really should never happen.

Reply:
We should fail fast if this is not suppose to happen to prevent weird behaviors. Most likely we would see permission denied in the launcher but it would be hard to correlate why(SELinux ?, permission bits? cgroup configuration?).

Reply:
Done.